### PR TITLE
Moulin initial: support moulin build

### PR DESCRIPTION
**Pull request by iusyk**
The domf,domd and dom0 have been adjusted to meet the prod-aos requirements.
Build is base at yocto 5.1, main stream is dunfell.
Supported machine is h3ulcb-ab
User domain DomF is supported by default.

Signed-off-by: Ihor Usyk <ihor_usyk@epam.com>

### CODE REVIEW

File: prod-aos-rcar.yaml [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+    desc: "Enable Multimedia support"
+    "no":
+      default: true
+      overrides:
+        components:
+          domd:
+            builder:
+              conf:
+                # Mask MMP recipes
+                - [BBMASK_append, "|kernel-module-uvcs-drv|omx-user-module"]
```
Comment (rshym) [inline prod-aos-rcar.yaml:380]: Please remove PREBUILT_DDK and ENABLE_MM, as not applicable to aos-rcar.

File: prod-aos-rcar.yaml [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+      domf_userdata:
+        gpt_type: E6D6D379-F507-44C2-A23C-238F2A3DF928 
+        type: ext4
+        size: 6022 MiB
+
+parameters:
+  # Machines
+  MACHINE:
+    desc: "RCAR Gen3-based device"
+    h3ulcb-ab:
```
Comment (rshym) [inline prod-aos-rcar.yaml:353]: Seems we need to add StarterKit 8GB as well.

File: prod-aos-rcar.yaml [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+        
+components:
+  dom0:
+    build-dir: "%{YOCTOS_WORK_DIR}"
+    default: true
+    sources:
+      - *COMMON_SOURCES
+      - type: git
+        url: "https://github.com/xen-troops/meta-xt-prod-aos-rcar.git"
+        rev: aos-base
```
Comment (rshym) [inline prod-aos-rcar.yaml]: Need to be replaced with `main`.

File: prod-aos-rcar.yaml [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+      - *COMMON_SOURCES
+      - type: git
+        url: https://github.com/xen-troops/meta-renesas.git
+        rev: "Renesas-Yocto-v5.1-patched"
+      - type: git
+        url: "git://git.yoctoproject.org/meta-selinux"
+        rev: dunfell
+      - type: git
+        url: "https://github.com/xen-troops/meta-xt-prod-aos-rcar.git"
+        rev: aos-base
```
Comment (rshym) [inline prod-aos-rcar.yaml]: Need to be replaced with `main`.

File: prod-aos-rcar.yaml [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+      - *COMMON_SOURCES
+      - type: git
+        url: "https://github.com/xen-troops/meta-renesas.git"
+        rev: "Renesas-Yocto-v5.1-patched"
+      - type: git
+        url: "git://git.yoctoproject.org/meta-selinux"
+        rev: dunfell
+      - type: git
+        url: "https://github.com/xen-troops/meta-xt-prod-aos-rcar.git"
+        rev: aos-base
```
Comment (rshym) [inline prod-aos-rcar.yaml]: Need to be replaced with `main`.

File: README.md [hunk 6]
```diff
@@ -1 +1,138 @@
-# meta-xt-prod-aos-rcar
\ No newline at end of file
+# meta-xt-prod-aos #
```
Comment (rshym) [inline README.md:1]: Please add `-rcar`

File: README.md [hunk 7]
```diff
@@ -1 +1,138 @@
...
+Then you can use `dd` to write this image to your SD card. Don't
+forget `conv=sparse` option for `dd` to speed up writing.
+
+### Using `rouge` in standalone mode
+
+In this mode you can write image right to SD card. But it requires
+additional options.
+
+In standalone mode`rouge` accepts the same parameters like
+`--MACHINE`, `--ENABLE_AOS_VIS` as `moulin` do.
```
Comment (rshym) [inline README.md:100]: Please remove `--ENABLE_AOS_VIS` option, as not available.

File: README.md [hunk 8]
```diff
@@ -1 +1,138 @@
...
+Those layers *may* be added and used manually, but they were written
+with [Moulin](https://moulin.readthedocs.io/en/latest/) build system,
+as Moulin-based project files provide correct entries in local.conf
+
+# Moulin project file
+
+Work is still in progress, but right now the following features are tested and working:
+
+* GPU sharing between domains
+* 3 domains are being built: Linux based Dom0, DomD and DomU
```
Comment (rshym) [inline README.md:18]: Please replace `DomU` with `DomF`.

File: README.md [hunk 9]
```diff
@@ -1 +1,138 @@
...
+
+Those layers *may* be added and used manually, but they were written
+with [Moulin](https://moulin.readthedocs.io/en/latest/) build system,
+as Moulin-based project files provide correct entries in local.conf
+
+# Moulin project file
+
+Work is still in progress, but right now the following features are tested and working:
+
+* GPU sharing between domains
```
Comment (rshym) [inline README.md:17]: Seems the GPU-related line needs to be removed.

File: README.md [hunk 10]
```diff
@@ -1 +1,138 @@
...
+* Networking in DomD and DomF
+* Network (NFS) boot for DomD and DomF
+* OP-TEE client in DomF
+* Virtualized OP-TEE build
+* ARM-TF that boots into EL2
+
+Features that are present but not tested:
+
+* AosBox with Starter Kit Premiere 8GB board
+* SD or eMMC boot
```
Comment (rshym) [inline README.md:28]: Maybe this need to be removed or moved to `tested and working` section.

File: prod-aos-rcar.yaml [hunk 11]
```diff
@@ -0,0 +1,387 @@
...
+variables:
+  YOCTOS_WORK_DIR: "yocto"
+  DOM0_BUILD_DIR: "build-dom0"
+  DOMD_BUILD_DIR: "build-domd"
+  DOMF_BUILD_DIR: "build-domf"
+  XT_DOMD_DTB_NAME: "%{SOC_FAMILY}-%{MACHINE}-domd.dtb"
+  XT_DOMU_DTB_NAME: "salvator-generic-domu.dtb"
+  XT_DOMA_DTB_NAME: ""
+  XT_XEN_DTB_NAME: "r8a77951-h3ulcb-4x2g-ab-xen.dtb"
+  XT_PVR_NUM_OSID: "2"
```
Comment (al1img) [inline prod-aos-rcar.yaml:12]: Do we need virtual GPU?

File: prod-aos-rcar.yaml [hunk 12]
```diff
@@ -0,0 +1,387 @@
...
+
+    # Remove features that we are not using
+    - [DISTRO_FEATURES_remove, "x11 gtk gobject-introspection-data wifi nfc bluetooth irda zeroconf 3g sysvinit"]
+
+domd_domu_conf: &DOMD_DOMU_CONF
+    - [MACHINE, "%{MACHINE}"]
+    - [SOC_FAMILY, "%{SOC_FAMILY}"]
+
+    # Add systemd configuration
+    - [DISTRO_FEATURES_append, " systemd"]
```
Comment (al1img) [inline prod-aos-rcar.yaml]: Why not add distro feature in one line as above?

File: prod-aos-rcar.yaml [hunk 13]
```diff
@@ -0,0 +1,387 @@
...
+        - [XT_GUEST_INSTALL, "%{XT_GENERIC_DOMU_TAG} %{XT_DOMA_TAG}"]
+        - [VIRTUAL-RUNTIME_init_manager, "systemd"]
+        # Machine feature 'multimedia' is used to enable (VSP in domd) and (LOSSY build option in ATF)
+        - [MACHINE_FEATURES_append, " multimedia"]
+        - [VIRTUAL-RUNTIME_init_manager, "systemd"]
+        - [BOARD_MODEL, "h3ulcb-ab;1.0"]
+        - [DOMD_IMAGE_VERSION, "2.0.0"]
+        - [DOMD_REF_VERSION, "1.0.0"]
+        - [BUNDLE_DOMD_TYPE, "full"]
+        - [INITRAMFS_IMAGE, "core-image-tiny-initramfs"]
```
Comment (al1img) [inline prod-aos-rcar.yaml:205]: Remove initramfs

File: prod-aos-rcar.yaml [hunk 14]
```diff
@@ -0,0 +1,387 @@
...
+        - [MACHINE_FEATURES_append, " multimedia"]
+
+        - [AOS_RUNNER, "crun"]
+        - [PREFERRED_PROVIDER_virtual/runc, "runc-opencontainers"]
+        # kernel provider
+        - [PREFERRED_PROVIDER_virtual/kernel, "linux-generic-armv8"]
+        # MACHINE
+        - [MACHINE, "generic-armv8-xt"]
+        #initramfs configuration
+        - [INITRAMFS_IMAGE, "core-image-tiny-initramfs"]
```
Comment (al1img) [inline prod-aos-rcar.yaml:274]: Remove initramfs

File: prod-aos-rcar.yaml [hunk 15]
```diff
@@ -0,0 +1,387 @@
...
+        - [INITRAMFS_IMAGE_BUNDLE, "0"]
+        - [INITRAMFS_FSTYPES, "ext4 cpio.gz"]
+        - [IMAGE_FSTYPES_append, " ext4"]
+        # Image version
+        - [DOMF_IMAGE_VERSION, "1.0.1"]
```
Comment (al1img) [inline prod-aos-rcar.yaml:279]: Remove image version.

File: prod-aos-rcar.yaml [hunk 16]
```diff
@@ -0,0 +1,387 @@
...
+        # Disable optee in meta-xt-images-domx layers
+        - [BBMASK_append, "meta-xt-images-domx/recipes-bsp/optee"]
```
Comment (al1img) [inline prod-aos-rcar.yaml:281]: Is it required?

File: meta-xt-prod-aos-rcar-control/recipes-guests/domu/domu.bbappend [hunk 17]
```diff
@@ -0,0 +1,48 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
```
Comment (al1img) [inline meta-xt-prod-aos-rcar-control/recipes-guests/domu/domu.bbappend:1]: Do we need domu folder?

File: meta-xt-prod-aos-rcar-domu/recipes-containers/crun/crun_%.bbappend [hunk 18]
```diff
@@ -0,0 +1,17 @@
+LICENSE = "GPLv2"
```
Comment (al1img) [inline meta-xt-prod-aos-rcar-domu/recipes-containers/crun/crun_%.bbappend:1]: remove crun

File: meta-xt-prod-aos-rcar-domu/recipes-core/initrdscripts/initramfs-framework_1.0.bbappend [hunk 19]
```diff
@@ -0,0 +1,4 @@
+# Use our initrd script instead of Yocto`s.
```
Comment (al1img) [inline meta-xt-prod-aos-rcar-domu/recipes-core/initrdscripts/initramfs-framework_1.0.bbappend:1]: remove iniramfs

File: meta-xt-prod-aos-rcar-domu/recipes-core/images/core-image-tiny-initramfs.bbappend [hunk 20]
```diff
@@ -0,0 +1,14 @@
+VIRTUAL-RUNTIME_dev_manager ?= "busybox-mdev"
```
Comment (al1img) [inline meta-xt-prod-aos-rcar-domu/recipes-core/images/core-image-tiny-initramfs.bbappend]: remove initramfs

File: meta-xt-prod-aos-rcar-domu/recipes-security/libseccomp/libseccomp_2.4.3.bb [hunk 21]
```diff
@@ -0,0 +1,46 @@
+SUMMARY = "interface to seccomp filtering mechanism"
```
Comment (al1img) [inline meta-xt-prod-aos-rcar-domu/recipes-security/libseccomp/libseccomp_2.4.3.bb]: remove libseccomp

File: prod-aos-rcar.yaml [hunk 22]
```diff
@@ -0,0 +1,387 @@
...
+  XT_DOMA_DTB_NAME: ""
+  XT_XEN_DTB_NAME: "r8a77951-h3ulcb-4x2g-ab-xen.dtb"
+  XT_PVR_NUM_OSID: "2"
+  XT_OP_TEE_FLAVOUR: "generic_dt"
+  XT_GENERIC_DOMU_TAG: "domf"
+  XT_DOMA_TAG: ""
+  SOC_FAMILY: "r8a7795"
+  XT_DOMD_CONFIG_NAME: "domd.cfg"
+  XT_DOMU_CONFIG_NAME: "domf.cfg"
+  XT_OP_TEE_FLAVOUR: "salvator_h3_4x2g"
```
Comment (al1img) [inline prod-aos-rcar.yaml:19]: XT_OP_TEE_FLAVOUR set above, which value is correct one?

File: prod-aos-rcar.yaml [hunk 23]
```diff
@@ -0,0 +1,387 @@
...
+  XT_DOMD_DTB_NAME: "r8a77951-h3ulcb-4x2g-ab-domd.dtb"
```
Comment (al1img) [inline prod-aos-rcar.yaml:20]: XT_DOMD_DTB_NAME set above, which value is correct one?

File: meta-xt-prod-aos-rcar-control/recipes-guests/doma/doma.bbappend [hunk 24]
```diff
@@ -0,0 +1,24 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
```
Comment (al1img) [inline meta-xt-prod-aos-rcar-control/recipes-guests/doma/doma.bbappend]: Do we need doma?

Comment (iusyk) [reply meta-xt-prod-aos-rcar-control/recipes-guests/doma/doma.bbappend]: no, removed

**Comment by iusyk** [comment]
not relevant

**Pull request closed without merging**